### PR TITLE
improve consistency for field names in mobilecoind API

### PR DESCRIPTION
**Pull request by rjwalters**
### Motivation

Our API currently uses different field names for the same data in different message types. This PR cleans up many of these problems.

### In this PR
* unifies "receiver", "recipient" and "public address" to "public address" (fixes MCC-1964)
* emphasizes "block count" and "block index"
* unifies "public_key" and "tx_public_key" as "tx_out_public_key"
* adds fee and tombstone where missing
* emphasizes when a subaddress's index is being used (as opposed to a subaddress's PublicAddress)
* avoids making any changes to layer below (lmdb objects) or to the mobilecoind-json API

### Future Work
* update the JSON version of these messages so that our API is maximally consistent MCC-1989
* update the python code MCC-1988
* update mobilecoind-mirror MCC1987
* handle fee and tombstone in GenerateOptimizationTx MCC-1991
* handle tombstone in GenerateTxFromTxOutList MCC-1992

### CODE REVIEW

File: mobilecoind-json/src/data_types.rs
Comment (eranrund) [inline mobilecoind-json/src/data_types.rs:456]: The field you're trying to convert is actually called `receiver` here.

File: mobilecoind-json/src/data_types.rs
Comment (eranrund) [inline mobilecoind-json/src/data_types.rs:1094]: This should actually be `get_tx_out_public_key` if we want to be specific. There's no public key for a transaction, only for a TxOut.

File: mobilecoind/api/proto/mobilecoind_api.proto
Comment (eranrund) [inline mobilecoind/api/proto/mobilecoind_api.proto:218]: Recommend `tx_out_public_key`

**Review by eranrund — APPROVED** [review]
Nice cleanup. Has some small nits/recommendation.
I suspect this might affect https://github.com/mobilecoinofficial/mobilecoind-mirror so after it gets merged can you please go there and uprev?

File: mobilecoind/api/proto/mobilecoind_api.proto
Comment (rjwalters) [reply mobilecoind/api/proto/mobilecoind_api.proto]: renamed throughout to `tx_out_public_key`

File: mobilecoind/api/proto/mobilecoind_api.proto
Comment (rjwalters) [inline mobilecoind/api/proto/mobilecoind_api.proto:650]: I think this might be better unchanged? @eranrund

Comment (eranrund) [reply mobilecoind/api/proto/mobilecoind_api.proto:650]: `receiver_b58_code` seems better to me

**Review by mfaulk — APPROVED** [review]

**Comment by sugargoat** [comment]
Let's hold off on merging this until we confirm a migration path for downstream users.

**Comment by sugargoat** [comment]
These are good changes, but we will need to rebase and rework, as well as confirm the migration path.

**Pull request closed without merging**
